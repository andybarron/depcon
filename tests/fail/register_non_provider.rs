use depcon::*;

#[derive(Injectable)]
struct NonProvider;

trait DummyService: 'static {}

fn main() {
    let mut container = Container::empty();
    container.register::<NonProvider, dyn DummyService>();
}
