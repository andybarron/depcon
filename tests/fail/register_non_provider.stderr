error[E0277]: the trait bound `NonProvider: Provider<(dyn DummyService + 'static)>` is not satisfied
   --> tests/fail/register_non_provider.rs:10:15
    |
10  |     container.register::<NonProvider, dyn DummyService>();
    |               ^^^^^^^^ the trait `Provider<(dyn DummyService + 'static)>` is not implemented for `NonProvider`
    |
note: required by a bound in `depcon::Container::register`
   --> src/container.rs
    |
    |         TProvider: Injectable + Provider<TService>,
    |                                 ^^^^^^^^^^^^^^^^^^ required by this bound in `depcon::Container::register`
